pkg/spec/deploymentconfig.go

```diff
@@ -0,0 +1,165 @@
+/*
+Copyright 2017 The Kedge Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package spec
+
+import (
+	log "github.com/Sirupsen/logrus"
+	"github.com/davecgh/go-spew/spew"
+	"github.com/ghodss/yaml"
+	os_deploy_v1 "github.com/openshift/origin/pkg/deploy/apis/apps/v1"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	kapi "k8s.io/kubernetes/pkg/api/v1"
+)
+
+// Unmarshal the Kedge YAML file
+func (deployment *DeploymentConfigSpecMod) Unmarshal(data []byte) error {
```

Comment:
can we change the receiver names here to `deploymentConfig` or `dc` or something, this is a bit confusing since it says `deployment`.

Reply:
done.